temporalio/client.py

```diff
@@ -4417,6 +4776,108 @@ async def terminate_workflow(self, input: TerminateWorkflowInput) -> None:
+                await self._client.data_converter.encode(input.args)
+            )
+        if input.headers is not None:
+            temporalio.common._apply_headers(
+                input.headers, req.request.input.header.fields
+            )
+        try:
+            resp = await self._client.workflow_service.update_workflow_execution(
+                req, retry=True, metadata=input.rpc_metadata, timeout=input.rpc_timeout
+            )
+        except RPCError as err:
+            # If the status is INVALID_ARGUMENT, we can assume it's an update
+            # failed error
+            if err.status == RPCStatusCode.INVALID_ARGUMENT:
+                raise WorkflowUpdateFailedError(input.workflow_id, input.update, err)
+            else:
+                raise
+
+        update_handle = WorkflowUpdateHandle(
+            client=self._client,
+            id=input.update_id,
+            name=input.update,
+            workflow_id=input.workflow_id,
+            run_id=input.run_id,
+            result_type=input.ret_type,
+        )
+        if resp.HasField("outcome"):
+            update_handle._set_known_result(resp.outcome)
+
+        return update_handle
+
+    async def poll_workflow_update(self, input: PollUpdateWorkflowInput) -> Any:
+        req = temporalio.api.workflowservice.v1.PollWorkflowExecutionUpdateRequest(
+            namespace=self._client.namespace,
+            update_ref=temporalio.api.update.v1.UpdateRef(
+                workflow_execution=temporalio.api.common.v1.WorkflowExecution(
+                    workflow_id=input.workflow_id,
+                    run_id=input.run_id or "",
+                ),
+                update_id=input.update_id,
+            ),
+            identity=self._client.identity,
+            wait_policy=temporalio.api.update.v1.WaitPolicy(
+                lifecycle_stage=temporalio.api.enums.v1.UpdateWorkflowExecutionLifecycleStage.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_COMPLETED
+            ),
+        )
+
+        async def poll_loop():
+            # Continue polling as long as we have either an empty response, or an *rpc* timeout
+            while True:
+                try:
+                    res = await self._client.workflow_service.poll_workflow_execution_update(
+                        req,
+                        retry=True,
+                        metadata=input.rpc_metadata,
+                        timeout=input.rpc_timeout,
+                    )
+                    if res.HasField("outcome"):
+                        return await _update_outcome_to_result(
+                            res.outcome,
+                            input.update_id,
+                            input.update,
+                            self._client.data_converter,
+                            input.ret_type,
+                        )
+                except RPCError as err:
+                    if err.status == RPCStatusCode.DEADLINE_EXCEEDED:
+                        continue
```

Comment:
```suggestion
                    if err.status != RPCStatusCode.DEADLINE_EXCEEDED:
                        raise
```

Hopefully this logic can go away soon when the server returns an empty outcome on their deadline exceeded